neonUtilities/R/byTileAOP.R

```diff
@@ -231,39 +193,58 @@ byTileAOP <- function(dpID, site="SJER", year="2017", easting, northing, buffer=
   } else {
     filepath <- paste(savepath, "/", dpID, sep="")
   }
-  if(dir.exists(filepath) == F) dir.create(filepath, showWarnings=F)
+  if(dir.exists(filepath) == F) {dir.create(filepath, showWarnings=F)}
 
   # copy zip files into folder
   j <- 1
   messages <- list()
   writeLines(paste("Downloading ", nrow(file.urls.current), " files", sep=""))
   pb <- utils::txtProgressBar(style=3)
   utils::setTxtProgressBar(pb, 1/(nrow(file.urls.current)-1))
+
+  counter<- 1
+
   while(j <= nrow(file.urls.current)) {
-    path1 <- strsplit(file.urls.current$URL[j], "\\?")[[1]][1]
-    pathparts <- strsplit(path1, "\\/")
-    path2 <- paste(pathparts[[1]][4:(length(pathparts[[1]])-1)], collapse="/")
-    newpath <- paste0(filepath, "/", path2)
-
-    if(dir.exists(newpath) == F) dir.create(newpath, recursive = T)
-    t <- try(downloader::download(file.urls.current$URL[j],
-                                  paste(newpath, file.urls.current$name[j], sep="/"),
-                                  mode="wb", quiet=T), silent = T)
-
-    if(class(t) == "try-error"){
-      writeLines("File could not be downloaded. URLs may have expired. Getting new URLs.")
-      file.urls.new <- getTileUrls(month.urls)
-      file.urls.current <- file.urls.new
-      writeLines("Continuing downloads.")}
-    if(class(t) != "try-error"){
-      messages[j] <- paste(file.urls.current$name[j], "downloaded to", newpath, sep=" ")
-      j = j + 1
+    counter<- counter + 1
+
+    if (counter > 2) {
+      stop(paste0("\nURL query for site (", site, ') and year (', year,
```

Comment:
Instead of stop(), use cat() for the message and then next. If one URL fails, we want to proceed to the next one on the list, not stop the function entirely.

Reply:
Sorry, forgot how this was structured. j <- j + 1 should work, not next.